### ISSUE TITLE
methylation rate not stable across different read positions in M-bias plot

### ISSUE THREAD

**Issue by hongenxu** [FelixKrueger/Bismark#97]
Hi Dr Krueger, 
in the M-bias plot (see file attached), it seems that methylation rate is not stable across read positions. I also noticed the page at https://sequencing.qcfail.com/articles/library-end-repair-reaction-introduces-methylation-biases-in-paired-end-pe-bisulfite-seq-applications/. Beside setting "--ignore_r2", what should I do? 

Thanks, 
Hongen

[example.pdf](https://github.com/FelixKrueger/Bismark/files/876100/example.pdf)

**Comment by FelixKrueger** [comment]
Hi Hongen,

Your data exhibits a fairly pronounced M-bias for at least 8bp in Read 1 (although this arguably nearly extends to base 20 or so) and also around 7-9bp for Read 2 at least. Just out of interest, was this data produced with the Accel Swift kit? We have tried to make a table with trimming recommendations here: https://github.com/FelixKrueger/Bismark/tree/master/Docs#viii-notes-about-different-library-types-and-commercial-kits

If I were you I would re-run the raw FastQ files through Trim Galore with the following options to remove the positions in question:

`trim_galore --paired --clip_r1 20 --clip_r2 15 BS-C19_R1.fastq.gz BS-C19_R2.fastq.gz`

And then run Bismark again using the same parameters. I hope this makes sense. Best, Felix

**Comment by hongenxu** [comment]
thanks for your help. I will come back after I get new M-bias plot. 
The library should be BSseq, and I will confirm.

**Comment by hongenxu** [comment]
it's TruSeq (EpiGnome).

**Comment by FelixKrueger** [comment]
Thanks for the update. Keeping your M-bias plots in mind I would still recommend the trimming options above. Cheers, Felix

**Comment by hongenxu** [comment]
Hi Felix,

see the file attached for updated M-bias plot. For read 2, there is a sharp increase in the end. Is it normal? What can I do? 
Best,
Hongen
[example_updated.pdf](https://github.com/FelixKrueger/Bismark/files/885275/example_updated.pdf)

**Comment by FelixKrueger** [comment]
Hi Hongen,

This looks much improved, the bias at the start seems to be almost gone. It is noteworthy though that you see a non-CG methylation level of 5-6% throughout. Is this an expected non-CG methylation or could it be indicative of a less-than-optimal bisulfite conversion efficiency?

The sharp spike at the end or Read 2 is expected. The reason for this is that the Illumina adapter sequence starts with `AGATC...` and Trim Galore is trimming very stringently to avoid any sort of adapter contamination. So if the last base of a read is `A` it will be removed, or if the last bases are `AG` or `AGA` they are removed etc. So this means that Read 2 may not end in a trailing A because it might be the start of the adapter sequence. But since Read 2 encodes methylation states as G and A (and not as C and T like  Read 1) it means that the last position of Read 2 cannot be an unmethylated residue.

We are aware that this behaviour may be somewhat biasing for the very last base of Read 2 of paired-end reads because it can only report one methylation state. This is ameliorated by the following facts:
- you would need a different read starting at a different position and ending at the very same position but with a different methylation state to see the bias at all (reads starting at the same position would get deduplicated)
- by the time you reach the very last position of Read 2 the information often overlaps with Read 1 and is therefore ignored by the option `--no_overlap` anyway.

In a nutshell: As long as you are not concerned about non-conversion issues I would be happy to proceed with the analysis of your data. Good luck!

**Comment by hongenxu** [comment]
Hi Felix, 

many thanks for your great explanations. I contacted my collaborator, and he said the non-CpG methylation is quite common in our dataset (placenta). 

Best, 
Hongen

**Comment by FelixKrueger** [comment]
Then you should be all set! Best, Felix
